spotlight-client/src/DataStore/UserStore.ts

```diff
@@ -121,6 +130,18 @@ export default class UserStore {
           this.isAuthorized = false;
           this.awaitingVerification = true;
         }
+        if (stateCode) {
+          this.stateCode = stateCode;
+          if (this.rootStore) {
+            this.rootStore.tenantStore.currentTenantId = stateCode;
+            // returning null renders an observable property immutable
+            intercept(
```

Comment:
It looks like MobX advises against using `intercept`, can you do the same thing with a reaction?

https://mobx.js.org/intercept-and-observe.html

Reply:
I saw this intercept being used in TenantStore's constructor, in order to prevent `currentTenantId` from changing. Maybe we don't need to do this, instead our code should just not change the `currentTenantId` if it's `locked`?

Reply:
I think this is fine and makes sense!

Reply:
Sorry, which part is fine, using the `intercept` or removing it?

Reply:
Using it! If it is already in this repo then might as well include it. I didn't realize public-dashboard was already using it.